app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -118,8 +118,9 @@
 
 public class NearbyParentFragment extends CommonsDaggerSupportFragment
         implements NearbyParentFragmentContract.View,
-        WikidataEditListener.WikidataP18EditListener, LocationUpdateListener {
+        WikidataEditListener.WikidataP18EditListener, LocationUpdateListener ,CheckBoxTriStates.Callback{
```

Comment:
I would expect zero changes to this file, if we just remove the state restoration in CheckboxTriState that is the minimal change to solve this issue

Reply:
@macgills  We have removed the manual state restoration thing, but the android system also restores the view right so if we don't set the callback in such cases, it will throw an NPE, right?

Reply:
This is called on master after onViewCreated

```
checkBoxTriStates.setCallback((o, state, b, b1) -> presenter.filterByMarkerType(o,state,b,b1));
```

There was an internal problem where the checkbox will throw an NPE if you don't set a click listener to it and it processes a click event, that should not be the case. That problem arose because the state restoration was happening before the click listener was set and your version of state restoration didn't use the `isRestoring` flag to prevent the state restoration being interpreted as human input.

Reply:
I am still halfway through my morning coffee, that wasn't a great explanation...

I'll perk up in a bit if you have any questions

Reply:
Fair point, done